test/parallel/test-assert-async.js

```diff
@@ -103,8 +103,9 @@ const invalidThenableFunc = () => {
   promises.push(assert.rejects(promise, {
     name: 'TypeError',
     code: 'ERR_INVALID_RETURN_VALUE',
+    // FIXME: This should use substring matching for key words, like /Promise/ and /undefined/
```

Comment:
I personally don't think we actually have to change this. It's only a single test and it's good to verify once that the outcome is fully as expected.

Reply:
I think if we do want that, it should be the a test for the errors (which don't currently exist), rather than a random test that happens to use that error (especially because the latter requires one to know which random test checks the full error message).

Reply:
nit: the usual format for this kind of comments in the code case looks more like:
```suggestion
    // TODO(JakobJingleheimer): This should use substring matching for key words, like /Promise/ and /undefined/
```
Alternatively we could have a test to validate the output of `determineSpecificType`, specific tests for the errors also sounds good to me.

Reply:
I used `FIXME` because there's an actual problem (whereas I see `TODO` as a tidy or improvement). Do we not differentiate?

Is the person cited in the parentheses an assignee or just the author? (I've been using it to assign things to myself).

I think a test of `determineSpecificType` would be appropriate / a good idea.

Reply:
> Is the person cited in the parentheses an assignee or just the author? (I've been using it to assign things to myself).

It's the author, so you can get pinged e.g. if someone else doesn't understand the comment.